sample-node-config/another-nixos/configuration.nix

```diff
@@ -0,0 +1,297 @@
+        if [ -f ${nodeId}-${name}.sk ]; then
+          echo "Found a ${name} key for ${nodeId}; not generating another one."
+          exit 0
+        fi
+
+        ${lib.getExe cardano-node.packages.${system}.cardano-cli} address key-gen \
+          --verification-key-file ${nodeId}-${name}.vk \
+          --signing-key-file ${nodeId}-${name}.sk
+      '';
+
+      genCardanoKey = genSomeCardanoKey "genCardanoKey" "node";
+      genFundsKey = genSomeCardanoKey "genFundsKey" "funds";
+    in
+    [
+      pkgs.git
+
+      # So you can just do (if you just want fresh credentials):
+      #
+      #  > cd ~/cardano-data/credentials
+      #  > genHydraKey
+      #  > genCardanoKey
+      #  > genFundsKey
+      #
+      genHydraKey
+      genCardanoKey
+      genFundsKey
+
+      hydra.packages."${system}".hydra-tui # To interact with your node/peers
+
+      # These aren't really needed, as the systemd services just pull in the
+      # binaries directly, but might be useful for debugging, so we leave them
+      # in the system path.
+      hydra.packages."${system}".hydra-node # To run a hydra node
+      cardano-node.packages."${system}".cardano-node # To talk to the cardano network
+      mithril.packages."${system}".mithril-client-cli # Efficient syncing of the cardano node
+      cardano-node.packages."${system}".cardano-cli # For any ad-hoc cardano actions we may like to run
+    ];
+
+  programs.bash.shellAliases = {
+    # Run 'logs -f' to follow
+    logs = "journalctl -u mithril-maybe-download -u cardano-node -u hydra-node -u necessary-files";
+
+    # Always open hydra-tui with the right args:
+    hydra-tui = "hydra-tui --testnet-magic ${networkMagic} --node-socket ${cardanoDataPath}/node.socket -k ${cardanoDataPath}/credentials/${nodeId}-funds.sk";
+  };
+
+  environment.variables = commonEnvVars;
+
+  systemd.services = {
+    necessary-files = {
+      requires = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+      wantedBy = [ "mithril-maybe-download.target" ];
+      path = with pkgs; [
+        curl
+        gnutar
+        gzip
+        git
+        cardano-node.packages."${system}".cardano-cli
+      ];
+      serviceConfig = {
+        User = "hydra";
+        Type = "notify";
+        NotifyAccess = "all";
+        ExecStart =
+          let
+            necessaryFiles = pkgs.writeShellScriptBin "necessaryFiles" ''
+              set -e
+
+              if [ -d ${cardanoDataPath} ]; then
+                echo "Not re-creating configs because ${cardanoDataPath} exists."
+                systemd-notify --ready
+                exit 0
+              fi
+
+              mkdir -p ${cardanoDataPath}/credentials
+
+              cd ${cardanoDataPath}
+
+              # Get the node configs
+              curl -L -O \
+                https://github.com/IntersectMBO/cardano-node/releases/download/${nodeVersion}/cardano-node-${nodeVersion}-linux.tar.gz
```

Comment:
Could: make this a flake input / (binary) nix derivation

Reply:
You mean the tar.gz file?

You can't get them from the flake input of the node, unfortunately, because files are not in the source; they are built as part of the release.

Reply:
See here: https://github.com/IntersectMBO/cardano-node/blob/master/nix/binary-release.nix#L54

Reply:
Maybe this would work - https://github.com/input-output-hk/cardano-configurations/tree/master/network/preview - but I'll stick with what I have for now :)